### PR TITLE
User can navigate the map by phone

### PR DESCRIPTION
**Pull request by reykjaviks**
with additional refactoring

### CODE REVIEW

File: client/src/common/navBar.tsx [hunk 1]
```diff
@@ -110,39 +221,104 @@ const NavBar = ({
...
+        <TopRightMenu>
+          <RightMenuItem>
+            <SearchBar placeholder="Search .." />
+          </RightMenuItem>
+
+          <RightMenuItem>
+            <Icon>
+              <TiLocationArrow />
+            </Icon>
+            <RightMenuText>Location Sharing</RightMenuText>
```
[the remark this answers is not in the document]
Comment (mriekkin) [reply client/src/common/navBar.tsx]: If you wish, you may do it like this: on your own branch (`git checkout feature/styles`) type `git merge master`. This brings you branch up-to-date, but doesn't modify the master branch. In this way, if you like, we can comment on the changes before doing the final merge.

File: client/src/router.tsx [hunk 2]
```diff
@@ -13,6 +13,9 @@ import MapContainer from './map/mapContainer';
 import AdminPanel, { RaspberryLocation } from './admin/adminPanel';
 import { Location } from './common/typeUtil';
 import NavBar from './common/navBar';
+
+import NavBar2 from './common/navBar2';
```
[the remark this answers is not in the document]
Comment (reykjaviks) [reply client/src/router.tsx]: OK

File: client/src/common/navBar.tsx [hunk 3]
```diff
@@ -66,36 +91,122 @@ const Content = styled.li`
...
 `;
 
-const AdminChip = styled.div`
-  font-size: 11px;
-  margin-right: 5px;
-  white-space: pre;
-  font-weight: 700;
-  color: white;
+const RightMenuText = styled.div`
+  /*margin-right: 5px;*/
```
[the remark this answers is not in the document]
Comment (reykjaviks) [reply client/src/common/navBar.tsx]: OK

File: client/src/common/navBar.tsx [hunk 4]
```diff
@@ -66,36 +91,122 @@ const Content = styled.li`
...
+  justify-content: flex-end;
+
+  @media (min-width: 575px) {
+    display: none;
+  }
+`;
+
+const Mobile = styled.nav<{ active: boolean }>`
+  height: auto;
+  display: inherit;
```
[the remark this answers is not in the document]
Comment (reykjaviks) [reply client/src/common/navBar.tsx]: I will look into it.

File: client/src/common/navBar.tsx [hunk 5]
```diff
@@ -66,36 +91,122 @@ const Content = styled.li`
...
   align-items: center;
-  cursor: pointer;
-  margin-right: 15px;
+  display: ${props => (props.active ? 'inherit' : 'none')};
+  -webkit-transition: width 2s, height 4s; /* For Safari 3.1 to 6.0 */
```
[the remark this answers is not in the document]
Comment (reykjaviks) [reply client/src/common/navBar.tsx]: OK

File: client/src/common/navBar.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (reykjaviks) [reply client/src/common/navBar.tsx]: OK will do.

**Pull request merged**